File: src/main/java/com/borvalcode/sudoku/infrastructure/gateway/sudokuboardapi/vo/SolveResponse.java
package com.borvalcode.sudoku.infrastructure.gateway.sudokuboardapi.vo;

import java.util.List;
import lombok.Data;

@Data
public class SolveResponse {

  private Response response;

  @Data
  public static class Response {

    private List<List<Integer>> board;

    private List<List<Integer>> solution;

    private boolean solvable;

    private String error;
  }
}

File: src/main/kotlin/com/borvalcode/sudoku/infrastructure/gateway/sudokuboardapi/vo/SolveResponse.kt
package com.borvalcode.sudoku.infrastructure.gateway.sudokuboardapi.vo

class SolveResponse(val response: Response) {

    class Response(val board: List<List<Int>>? = null,
                   val solution: List<List<Int>>,
                   val solvable: Boolean = false,
                   val error: String? = null)
}